gradoop-flink/src/main/java/org/gradoop/flink/algorithms/gelly/vertexdegrees/DistinctVertexDegrees.java

```diff
@@ -0,0 +1,105 @@
+import org.gradoop.flink.algorithms.gelly.GellyAlgorithm;
+import org.gradoop.flink.algorithms.gelly.functions.EdgeToGellyEdgeWithNullValue;
+import org.gradoop.flink.algorithms.gelly.functions.VertexToGellyVertexWithNullValue;
+import org.gradoop.flink.algorithms.gelly.vertexdegrees.functions.DistinctVertexDegreesToAttribute;
+import org.gradoop.flink.model.api.epgm.LogicalGraph;
+import org.gradoop.flink.model.impl.functions.epgm.Id;
+
+/**
+ * A gradoop operator wrapping {@link org.apache.flink.graph.asm.degree.annotate.directed.VertexDegrees}.
+ */
+public class DistinctVertexDegrees extends GellyAlgorithm<NullValue, NullValue> {
+
+  /**
+   * Property key to store the sum vertex degree in.
+   */
+  private final String propertyKey;
+  /**
+   * Property key to store the in vertex degree in.
+   */
+  private final String propertyKeyIn;
+  /**
+   * Property key to store the out vertex degree in.
+   */
+  private final String propertyKeyOut;
+
+  /**
+   * Output vertices with an in-degree of zero.
+   */
+  private final boolean includeZeroDegreeVertices;
+
+  /**
+   * Constructor for Vertex Degree with in- and out-degree and total of degrees of a graph.
+   * Note: The Gelly implementation count loops between edges like (v1) -> (v2),
+   * (v2) -> (v1) as one.
+   *
+   * @param propertyKey Property key to store the sum of in- and out-degrees in.
+   * @param propertyKeyIn Property key to store the in-degree in.
+   * @param propertyKeyOut Property key to store the out-degree in.
+   */
+  public DistinctVertexDegrees(String propertyKey, String propertyKeyIn, String propertyKeyOut) {
+    this(propertyKey, propertyKeyIn, propertyKeyOut, true);
+  }
+
+  /**
+   * Constructor for Vertex Degree with fixed set of whether to output
+   * vertices with an in-degree of zero.
+   *
+   * @param propertyKey Property key to store the sum of in- and out-degrees in.
+   * @param propertyKeyIn Property key to store the in-degree in.
+   * @param propertyKeyOut Property key to store the out-degree in.
+   * @param includeZeroDegreeVertices whether to output vertices with an
+   *                                  in-degree of zero
+   */
+  public DistinctVertexDegrees(String propertyKey, String propertyKeyIn, String propertyKeyOut,
+    boolean includeZeroDegreeVertices) {
+    super(new VertexToGellyVertexWithNullValue(), new EdgeToGellyEdgeWithNullValue());
+    this.propertyKey = propertyKey;
+    this.propertyKeyIn = propertyKeyIn;
+    this.propertyKeyOut = propertyKeyOut;
+    this.includeZeroDegreeVertices = includeZeroDegreeVertices;
+  }
+
+  @Override
+  protected LogicalGraph executeInGelly(Graph<GradoopId, NullValue, NullValue> graph)
+    throws Exception {
+    DataSet<Vertex> newVertices =
+      new org.apache.flink.graph.asm.degree.annotate.directed.VertexDegrees<GradoopId, NullValue,
+      NullValue>()
+      .setIncludeZeroDegreeVertices(includeZeroDegreeVertices)
+      .run(graph).join(currentGraph.getVertices())
```

Comment:
change to 
```
.run(graph)
.join(currentGraph.getVertices())
.where(0).equalTo(new Id<>())
.with(new DistinctVertexDegreesToAttribute(propertyKey, propertyKeyIn, propertyKeyOut));
```

Reply:
Thanks for the feedback, I will change that.